### ISSUE TITLE
Attempted to wrap ... which is already wrapped

### ISSUE THREAD

**Issue by hellboy81** [wallabyjs/public#194]
- First Run: no error
- some code changed, tests rerun, repeat
- after some code changed - error:
   ![screenshot](http://pasteboard.co/1VyNsMDd.png)
- restart walaby - no error

**Comment by ArtemGovorov** [comment]
Attached screenshot link is broken, please paste another screenshot.

**Comment by sertae** [comment]
I hope I'm not hijacking this thread. I had same error over and over again with Sinon and I was about to ask what I'm doing wrong. So maybe I can add some details.

TL;DR; Setting workers: { recycle: true } fixes this.

But I'm not sure if it's a good solution. Documentation says it's not necessary if you're cleaning up after your tests. I think I do, but I'm new to testing (thanks to Wallaby :+1: ) so it's time to ask for help.

I've read your Sails.js example so I assume you are familiar with Sails. I've got two separate describes with Sinon stubs for Sails custom service. One for success and the other for error.

	describe('image resize success',function () {

		var resize;

		before(function () {
			resize = sinon.stub(app.services.image, 'resize', function (options, next) {
				return next(null, 'tests/mocks/minions1.jpg');
			});
		});

		after(function () {
			resize.restore();
		});

		it('should return resized file with correct width and height', function (done) {
			var res = {
				send: sinon.spy(function () {
					expect(res.send).to.have.been.called;
					done();
				}),
			};

			ImageController.resize(null, res);
		});
	});
	
	describe('image resize error',function () {

		var resize;

		before(function () {
			resize = sinon.stub(app.services.image, 'resize', function (options, next) {
				return next('fail');
			});
		});

		after(function () {
			resize.restore();
		});

		it('should return resized file with correct width and height', function (done) {
			var res = {
				serverError: sinon.spy(function () {
					expect(res.serverError).to.have.been.called;
					done();
				})
			};

			ImageController.resize(null, res);
		});
	});

I think that somewhere in between they are called in the wrong order and that's how "Attempted to wrap ... which is already wrapped" error shows up:

```
before()
it()
before() // not restored yet, second sinon.stub is called? "Attempted to wrap ... which is already wrapped" 
it()
after() // unwrap
after() // already restored, another error: "Restore is not a function"
```

As original report stated this does not happen at first Wallaby run and on restarts. Restart is often necessary to fix the issue. It works few times, code can be changed without an error and when it happens Wallaby restart is the only option left.

`workers: { recycle: true }` is currently the solution I've found. But maybe you will have better idea. Is there something Wallaby or I can do better with my tests?

**Comment by hellboy81** [comment]
I have problems with stubbing of global Date obejct with sinon sandbox.

**Comment by ArtemGovorov** [comment]
Thanks for providing the way to reproduce it. The issue is fixed in core v1.0.83.
